server/store/user_store.go

```diff
@@ -278,3 +288,86 @@ func (index UserIndex) GetMattermostUserIDs() []string {
 
 	return result
 }
+
+// Get the check user status function
+func GetCheckUserStatus(store Store, logger bot.Logger, mattermostUserID string) func() bool {
+	return func() bool {
+		user, err := store.LoadUser(mattermostUserID)
+		if err != nil {
+			logger.Errorf("Not able to load the user %s. error: %s", mattermostUserID, err.Error())
+			return false
+		}
+
+		// Check if the user is marked as inactive
+		if user.OAuth2Token == nil {
+			return false
+		}
+
+		return true
+	}
+}
+
+// Get the change user status function
+func GetChangeUserStatus(store Store, logger bot.Logger, mattermostUserID string, poster bot.Poster) func(error) {
+	return func(err error) {
```

Comment:
This `return func` pattern is kind of awkward, but I suppose makes sense in order to encapsulate the passed in args